pkg/kdump/config-client.js

```diff
@@ -113,21 +124,185 @@ export class ConfigFile {
+            }
+        });
+
+        // default to local if no target configured
+        if (Object.keys(this.settings.targets).length === 0)
+            this.settings.targets.local = { type: "local", path: path.value };
+
+        // only allow compression if there is no core collector set or it's set to makedumpfile
+        this.settings.compression.allowed = (
+            !("core_collector" in this.settings._internal) ||
+            (this.settings._internal.core_collector.value.trim().indexOf("makedumpfile") === 0)
+        );
+        // compression is enabled if we have a core_collector command with the "-c" parameter
+        this.settings.compression.enabled = (
+            ("core_collector" in this.settings._internal) &&
+            this.settings._internal.core_collector.value &&
+            (this.settings._internal.core_collector.value.split(" ").indexOf("-c") != -1)
+        );
+    }
+
+    /* update single _internal setting to given value
+     * make sure setting exists if value is not empty
+     */
+    _updateSetting(settings, key, value) {
+        if (key in settings._internal) {
+            if (value)
+                settings._internal[key].value = value;
+            else
+                delete settings._internal[key];
+        } else {
+            if (value)
+                settings._internal[key] = { value: value };
+        }
+    }
+
+    /* transform settings from model back to _internal format
+     * this.settings = current state from file
+     * settings = in-memory state from UI
+     */
+    _persistSettings(settings) {
+        // target
+        if (Object.keys(settings.targets).length > 0) {
+            const target = Object.values(settings.targets)[0];
+            this._updateSetting(settings, "path", target.path);
+
+            // wipe old target settings
+            for (const key in this.settings.targets) {
+                const oldTarget = this.settings.targets[key];
+                if (oldTarget.type == "mount") {
+                    delete settings._internal[oldTarget.fsType];
+                } else if (oldTarget.type == "ssh") {
+                    delete settings._internal.ssh;
+                    delete settings._internal.sshkey;
+                } else {
+                    delete settings._internal[key];
+                }
+            }
+
+            if (target.type === "nfs") {
+                this._updateSetting(settings, "nfs", [target.server, target.export].join(":"));
+            } else if (target.type === "ssh") {
+                this._updateSetting(settings, "ssh", target.server);
+                if ("sshkey" in target)
+                    this._updateSetting(settings, "sshkey", target.sshkey);
+            } else if (target.type === "raw") {
+                this._updateSetting(settings, "raw", target.partition);
+            } else if (target.type === "mount") {
+                this._updateSetting(settings, target.fsType, target.partition);
+            }
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17544-20220719-143653-7a0d5780-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)